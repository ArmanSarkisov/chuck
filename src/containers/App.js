import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from 'react-router-dom';
import { ToastContainer } from 'react-toastify';

// private route
import { PrivateRoute } from './PrivateRoute';

// pages
import { Jokes } from '../pages/Jokes';
import { Login } from '../pages/auth/Login';
import { Favorites } from '../pages/Favorites';
import { Header } from '../components/Header';
import { Register } from '../pages/auth/SignUp';

function App() {
  return (
    <Router>
      <div className="App">
        <Header />
        <ToastContainer />
        <Switch>
          <Route exact path="/jokes" component={Jokes} />
          <Route exact path="/auth/login" component={Login} />
          <Route exact path="/auth/sign-up" component={Register} />
          <PrivateRoute exact path="/favorites" component={Favorites} />
          <Redirect from="*" to="/auth/login" />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
