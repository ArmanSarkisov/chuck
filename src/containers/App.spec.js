import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe('App component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<App />);
  });

  it('should render Joke component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
