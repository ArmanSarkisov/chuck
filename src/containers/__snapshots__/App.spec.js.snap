// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render Joke component 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Header />
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggableDirection="x"
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      transition={[Function]}
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/jokes"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/auth/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/auth/sign-up"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/favorites"
      />
      <Redirect
        from="*"
        to="/auth/login"
      />
    </Switch>
  </div>
</BrowserRouter>
`;
